New pillow release has broken array coercion

## Description

Probably an issue for Pillow, but not sure yet so noting it down here. Example failure [here](https://dev.azure.com/scikit-image/scikit-image/_build/results?buildId=6203&view=logs&j=6ce3b3d3-5fdf-5465-5577-d7690203f228&t=ea6beca8-6c3f-5e5a-4621-92d65119556e&l=6832):

```pytb
>               frame_paletted = np.array(im, np.uint8)
E               TypeError: __array__() takes 1 positional argument but 2 were given

C:\hostedtoolcache\windows\Python\3.9.5\x64\lib\site-packages\imageio\plugins\pillow.py:745: TypeError
```

I'm pretty sure that `__array__` methods *should* take in `dtype=`, but I don't know 100%.

--- a/requirements/README.md
+++ b/requirements/README.md
@@ -38,5 +38,6 @@ $ pip install -U -r requirements/test.txt
 * matplotlib 3.0.0 is not used because of a bug that collapses 3D axes (see https://github.com/scikit-image/scikit-image/pull/3474 and https://github.com/matplotlib/matplotlib/issues/12239).
 * pillow 7.1.0 fails on png files, See https://github.com/scikit-image/scikit-image/issues/4548
 * pillow 7.1.1 fails due to https://github.com/python-pillow/Pillow/issues/4518
+* pillow 8.3.0 broke array coercion when dtype was specified https://github.com/python-pillow/Pillow/pull/5572
 * imread 0.7.2 fails due to build failure https://github.com/luispedro/imread/issues/36
 * sphinx-gallery 0.8.0 is banned due to bug introduced on binder: https://github.com/scikit-image/scikit-image/pull/4959#issuecomment-687653537

--- a/requirements/default.txt
+++ b/requirements/default.txt
@@ -2,7 +2,7 @@ numpy>=1.17.0
 scipy>=1.2.3
 matplotlib>=3.0.3
 networkx>=2.2
-pillow>=6.1.0,!=7.1.0,!=7.1.1
+pillow>=6.1.0,!=7.1.0,!=7.1.1,!=8.3.0
 imageio>=2.4.1
 tifffile>=2019.7.26
 PyWavelets>=1.1.1

